server/src/test/java/io/spine/server/storage/given/RecordStorageTestEnv.java

```diff
@@ -0,0 +1,185 @@
+
+import static io.spine.server.entity.TestTransaction.injectState;
+import static io.spine.server.entity.storage.EnumType.STRING;
+import static io.spine.util.Exceptions.illegalStateWithCauseOf;
+
+/**
+ * @author Dmytro Dashenkov
+ * @author Dmytro Kuzmin
+ */
+public class RecordStorageTestEnv {
+
+    /** Prevents instantiation of this utility class. */
+    private RecordStorageTestEnv() {
+    }
+
+    @SuppressWarnings("unused") // Reflective access
+    public static class TestCounterEntity<I> extends TransactionalEntity<I,
+            Project,
+            ProjectVBuilder> {
+
+        private int counter = 0;
+
+        public TestCounterEntity(I id) {
+            super(id);
+        }
+
+        @CanIgnoreReturnValue
+        @Column
+        public int getCounter() {
+            counter++;
+            return counter;
+        }
+
+        @Column
+        public long getBigCounter() {
+            return getCounter();
+        }
+
+        @Column
+        public boolean isCounterEven() {
+            return counter % 2 == 0;
+        }
+
+        @Column
+        public String getCounterName() {
+            return getId().toString();
+        }
+
+        @Column(name = "COUNTER_VERSION" /* Custom name for storing
+                                            to check that querying is correct. */)
+        public Version getCounterVersion() {
+            return Version.newBuilder()
+                          .setNumber(counter)
+                          .build();
+        }
+
+        @Column
+        public Timestamp getNow() {
+            return Time.getCurrentTime();
+        }
+
+        @Column
+        public Project getCounterState() {
+            return getState();
+        }
+
+        @Column
+        public int getProjectStatusValue() {
+            return getState().getStatusValue();
+        }
+
+        @Column
+        public ProjectStatus getProjectStatusOrdinal() {
+            return Enum.valueOf(ProjectStatus.class, getState().getStatus().name());
+        }
+
+        @Column
+        @Enumerated(STRING)
+        public ProjectStatus getProjectStatusString() {
+            return Enum.valueOf(ProjectStatus.class, getState().getStatus().name());
+        }
+
+        public void assignStatus(Project.Status status) {
+            final Project newState = Project.newBuilder(getState())
+                                            .setStatus(status)
+                                            .build();
+            injectState(this, newState, getCounterVersion());
+        }
+
+        public void archive() {
+            TestTransaction.archive(this);
+        }
+
+        public void delete() {
+            TestTransaction.delete(this);
+        }
+    }
+
+    /**
+     * The {@link TestCounterEntity} {@linkplain Project.Status project status} represented by the
+     * {@linkplain Enum Java Enum}.
+     *
+     * <p>Needed to check filtering by the {@link Enumerated} entity columns.
```

Comment:
`It is used to test the entity filtering when the query is built around `{@link Enumerated} entity columns.`

However, I am not sure why we don't use the Java enum generated from the Protobuf definition, but instead we duplicate the definition.